src/bin/ambit/cmd.rs

```diff
@@ -316,3 +407,94 @@ pub fn git(arguments: Vec<&str>) -> AmbitResult<()> {
     io::stdout().write_all(&output.stderr)?;
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::get_paths_from_spec;
+    use ambit::config::ast::Spec;
+    use std::{
+        collections::HashSet,
+        fs::{self, File},
+        path::PathBuf,
+    };
+
+    fn test_spec(spec_str: &str, existing_paths: &[&str], expected_paths: &[PathBuf]) {
+        let spec = Spec::from(spec_str);
+        let dir_path = tempfile::tempdir().unwrap().into_path();
+        // Create paths.
+        for path in existing_paths {
+            let path = dir_path.join(path);
+            if let Some(parent) = path.parent() {
+                fs::create_dir_all(parent).unwrap();
+            }
+            File::create(path).unwrap();
+        }
+        let paths = get_paths_from_spec(&spec, dir_path).unwrap();
+        let paths: HashSet<&PathBuf> = paths.iter().collect();
+        // Use a HashSet as order of paths should not matter.
```

Comment:
Duplicates should not be silently discarded; they should be checked to make sure they don't slip in.